downloads/swan-lake-release-notes/swan-lake-beta3/RELEASE_NOTE.md

```diff
@@ -0,0 +1,780 @@
+       self.'start = self.engine.'start;
+   }
+}
+ 
+public function main() {
+   Car car = new;
+   car.'start(); // Call the function via the object field.
+}
+```
+
+#### Tuple to JSON Compatibility
+
+A tuple value whose members are JSON compatible can now be used in a context that expects a JSON value.
+
+```ballerina
+[string, int, boolean...] a = ["text1", 1, true, false];
+// Now allowed.
+json b = a;
+```
+
+#### Error Return in the `init` Method of a Service Declaration
+
+Previously, the `init` method of a service declaration could not have a return type containing an error. That restriction has been removed with this release.
+If the `init` method of a service declaration returns an error value, it will result in a module initialization failure.
+
+```ballerina
+import ballerina/http;
+ 
+service / on new http:Listener(8080) {
+   function init() returns error? {
+       // Return an error for demonstration.
+       // This will result in module initialization failing.
+       return error("Service init failure!");
+   }
+}
+```
+
+#### Using `check` in Object Field Initializers
+
+`check` can now be used in the initializer of an object field if the class or object constructor expression has an `init` method with a compatible return type (i.e., the error type that the expression could evaluate to is a subtype of the return type of the `init` method).
+
+If the expression is used with `check` results in an error value, the `init` method will return the error resulting in either the `new` expression returning an error or the object constructor expression resulting in an error.
+
+```ballerina
+import ballerina/io;
+ 
+int? value = ();
+ 
+class NumberGenerator {
+   int lowerLimit;
+ 
+   // `check` can be used in the field initializer
+   // since the `init` method's return type allows `error`.
+   int upperLimit = check value.ensureType();
+ 
+   function init(int lowerLimit) returns error? {
+       self.lowerLimit = lowerLimit;
+   }
+}
+ 
+public function main() {
+   NumberGenerator|error x = new (0);
+ 
+   if x is error {
+       io:println(x);
+   }
+}
+```
+
+#### Wildcard Binding Pattern Support in Variable Declarations
+
+The wildcard binding pattern can now be used in a variable declaration with a value that belongs to type `any`.
+
+```ballerina
+import ballerina/io;
+ 
+float _ = 3.14;
+var _ = io:println("hello");
+```
+
+Using the wildcard binding pattern when the value is an error will result in a compilation error.
+
+```ballerina
+var _ = error("custom error"); // Compilation error.
+```
+
+#### Relaxed Static Type Requirements for `==` and `!=`
+
+Previously, `==` and `!=` were allowed only when both operands were of static types that are subtypes of `anydata`. This has now been relaxed to allow `==` and `!=` if the static type of at least one operand is a subtype of `anydata`.
+
+```ballerina
+error? x = ();
+ 
+// Now allowed.
+if x == () {
+ 
+}
+``` 
+
+#### Relaxed Query Expression Keywords
```

Comment:
@hasithaa, @rdulmina, since the spec issue for this is still open, should we avoid mentioning this specifically in the release note for beta3? In case there could be some changes?

Reply:
Spec Issue is a more general one right? This is just a part of that issue and I think this is finalized. Hence, IMO this information is worth mentioning in the release note.

Reply:
Shall we remove this from the release note for now.